pkg/cluster/internal/create/nodes.go

```diff
@@ -206,11 +206,26 @@ func nodesToCreate(cfg *config.Cluster, clusterName string) []nodeSpec {
 				controlPlaneImage = configNode.Image
 			}
 		}
+	}
+	isHA := controlPlanes > 1
+
+	// add all of the config nodes as desired nodes
+	for _, configNode := range configNodes {
+		role := string(configNode.Role)
+		apiServerPort := cfg.Networking.APIServerPort
+		apiServerAddress := cfg.Networking.APIServerAddress
+		// only the external LB should reflect the port if we have
+		// multiple control planes
+		if isHA && role != constants.ExternalLoadBalancerNodeRoleValue {
+			apiServerPort = 0
```

Comment:
so actually this should also probably set `apiServerAddress` to `127.0.0.1`, only the single CP or LB needs to be listening on the configured addr and port, the rest should be local + random 🤔

Reply:
hm, i think you are right about this.
cc @fabriziopandini 

btw, `isHA` should probably be `hasMultipleCP` or something like that.
because HA is a bit ambiguous, e.g. HA is at least 3 CPs, behind a LB.

Reply:
or really `wantLoadBalancer`, since we may eventually unblock an option to have multiple CP but no LB at some point, we'll need to think these things through for future config options